src/status_im2/contexts/add_new_contact/events.cljs

```diff
@@ -6,114 +6,133 @@
             [status-im.ethereum.ens :as ens]
             [status-im.ethereum.stateofus :as stateofus]
             [status-im.native-module.core :as status]
-            [status-im2.navigation.events :as navigation]
-            [status-im2.utils.validators :as validators]
+            [status-im2.utils.validators :as v]
             [status-im.utils.utils :as utils]))
 
+;; 1. Domain model
+;; - start with empty 'schema' and progressively add/update to it
+;; - persist this data structure to app-db
+;; 2. Composability https://www.youtube.com/watch?v=kuNxVXnmjHA
+;; - pure functions that transform maps
+;; 3. Tests
+;; - focus on the core decision-making functions first (validate-contact)
+;; - followed by the (hopefully mostly empty) event handlers
+;; - followed by int test (add-contact-test) for status-go & downstream events
+;; 4. Testing event handlers/callbacks
+;; - remove all logic from callback: callback only dispatches
+;; - tests replace the callback with a fn that just returns args
+
+(defn init-contact ; in app-db as [:contacts/new-identity]
+  ([]
+   {:user-public-key nil ; user's public key (not the contact)
```

Comment:
perhaps this is better as a docstring for consistency as I don't think we have much other commenting formats like this. 🤔

Reply:
Maybe I'm biased, but I sort of like this approach, with the margin comments close to the actual map.

Reply:
@J-Son89 @ilmotta I'm on the fence here, I would have just converted it to a docstring but Icaro mirrored my own thinking here, which is that juxtaposing the comment right next to the actual key makes it more likely that it won't be overlooked when the args inevitably change (it's on the same line, hard to ignore). This is a small thing for me, so no worrries either way.

You know, this could also be data in the map (the description, the set of possible values), WDYT?

Reply:
Im okay with the change but if it's a better practice then we could make it the standard.
More so it highlights the need for some real typing/spec implementation.
Which I believe we are slowly working towards

Reply:
Tangent here: @J-Son89, remember that discussion about docstrings and spec in a separate PR? Well, I made some serious progress in all things related to specs for status-mobile, but using Malli. I kindly ask for your patience and the rest of the team, it might take 2-3 weeks to get a solid implementation and I'll run a presentation as well, because there are so many interesting things to talk about.

Reply:
Yep I recall and that's what I was referring to!

Reply:
Converted to docstring. Thanks for the helpful feedback!